astropy/convolution/boundary_none.pyx

```diff
@@ -82,7 +94,15 @@ def convolve2d_boundary_none(np.ndarray[DTYPE_t, ndim=2] f,
     cdef np.ndarray[DTYPE_t, ndim=2] conv = np.zeros([nx, ny], dtype=DTYPE)
 
     cdef unsigned int i, j, ii, jj
-
+    cdef unsigned int nkx_minus_1 = nkx-1, nky_minus_1 = nky-1
```

Comment:
These loop vars and their corresponding computations maybe hoisted by the C compiler, however, it is better to be explicit and hoist them to their correct positions manually.
I say maybe instead of probably because the code is not straight C and some of the vars, i.e. `nkx` & `wky` cannot be declared as `const` (not possible in Cython - to my knowledge) and therefore probably won't be constant-folded and possibly not even hoisted.